website/docs/guides/custom-components.mdx

```diff
@@ -79,33 +79,43 @@ For example, you can use a custom [DayButton](../api/functions/DayButton.md) to
 ```tsx title="./MyDatePicker.tsx"
 import React from "react";
 
-import { DayPicker } from "react-day-picker";
+import { DayButtonProps, DayPicker } from "react-day-picker";
 
-export function MyDatePicker() {
-  const [selected, setSelected] = React.useState<Date>();
+const SelectedDateContext = React.createContext<{
+  selected?: Date;
+  setSelected?: React.Dispatch<React.SetStateAction<Date | undefined>>;
+}>({});
+
+function DayButton(props: DayButtonProps) {
+  const { day, modifiers, ...buttonProps } = props;
+
+  const { setSelected } = React.use(SelectedDateContext);
```

Comment:
There should be no need to use a new context to store the selected state. DayPicker already runs inside a similar context, which is accessible through the [DayPicker hook](https://daypicker.dev/guides/custom-components#daypicker-hook).

```suggestion
  const { select } = useDayPicker();
```

Reply:
I tried to use it, but we have some caveats to deal with it:
<img width="445" alt="image" src="https://github.com/user-attachments/assets/89de5359-85bb-449c-8921-c6b894a361b9" />

It doesn't accept `undefined` as an argument for the `Date`.
Also each `select` based on the `mode` has specific logic related to them that the consumers may not be aware.
E.g. the `select` logic for `single` day `mode`: https://github.com/gpbl/react-day-picker/blob/main/src/selection/useSingle.tsx

Let me know if we should rethink this example.
I tried to keep the same logic in this PR and focus on the docs update, that's why I avoiding changing a lot the example.